content/src/main/content/jcr_root/apps/acs-commons/components/utilities/qr-code/clientlibs/js/app.js

```diff
@@ -0,0 +1,134 @@
+/*
+ * #%L
+ * ACS AEM Commons Package
+ * %%
+ * Copyright (C) 2014 Adobe
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L% 
+ */
+
+/*global angular: false, ace: false qrCode: false*/
+
+angular.module('acs-commons-qr-code-app', ['acsCoral', 'ACS.Commons.notifications']).controller('MainCtrl', ['$scope', '$http', '$timeout', 'NotificationsService', function ($scope, $http, $timeout, NotificationsService) {
+
+    $scope.app = {
+        uri: ''
+    };
+
+    $scope.form = {
+        properties: [{
+                name: '',
+                value: ''
+            }
+        ]
+    };
+
+    $scope.results = {};
+
+    $scope.addProperty = function (properties) {
+        properties.push({
+            name: '',
+            value: ''
+        });
+    };
+
+    $scope.removeProperty = function (properties, index) {
+        properties.splice(index, 1);
+    };
+
+    $scope.saveConfig = function () {
+
+        // If enabled, at least 1 entry should be present
+        var isValid = function () {
+            return $scope.form.properties.length >= 1 && $scope.form.properties[0].name !== "" && $scope.form.properties[0].value !== "";
+
+        };
+
+        if ($scope.form.enable && !isValid()) {
+            NotificationsService.add('error', "Error", "Please add a QR configuration");
+            return;
+        }
+
+        $http({
+            method: 'POST',
+            url: $scope.app.uri,
+            data: 'config=' + encodeURIComponent(JSON.stringify($scope.form)),
+            headers: {
+                'Content-Type': 'application/x-www-form-urlencoded'
+            }
+        }).
+        success(function (data, status, headers, config) {
+            $scope.app.running = false;
+            NotificationsService.add('success', "Success", "Your configuration has been saved");
+            NotificationsService.running($scope.app.running);
+
+            if ($scope.form.enable) {
+                $http({
```

Comment:
@davidjgonzalez Possibly I'm missing something and need to test this locally, but at first glance... wouldn't this category property get blown away on the next deployment of ACS AEM Commons?